### PR TITLE
Fixup docker support

### PR DESCRIPTION
**Pull request by yebyen**
Add coreutils to `docker-support/Dockerfile`

* some other changes related to making it a bit easier for maintainers to keep this up to date.

If a maintainer can run `make docker-push-support` that is the quickest way to bring the image current, when this gets merged.

The errors this fixes are related to recent changes in support scripts:

    * tail --lines=+2 in hack/import-flux2-assets.sh

(Fuller/more complete details are in the commit message)

### CODE REVIEW

**Comment by dholbach** [comment]
This largely looks good to me, but I won't be able to push the image.

With #266 the `tail --lines=+2` bit will be gone.

**Comment by yebyen** [comment]
Maybe we don't need coreutils then

Let's leave it open until it resolves itself one way or another, (we
shouldn't merge it without doing the update at the same time)

The makefile refactoring and the documentation updates still make sense to
include even without the coreutils update

👍

File: docker-support/Dockerfile [hunk 1]
```diff
@@ -8,6 +8,7 @@ RUN apk update && \
 	make \
 	nodejs \
 	npm \
+	coreutils \
```
Comment (dholbach) [inline docker-support/Dockerfile:11]: #266 has been merged, so I assume we won't need this - but it won't hurt either I assume...

Comment (kingdonb) [reply docker-support/Dockerfile:11]: Next time someone tries to use something advanced and unsupported by busybox for any command, we won't trip over it again. 👍 I'm not sure anyone would have noticed but I just made this and so, search and destroy error messages.

There is no urgency to merge this or to update the website support image.

**Comment by kingdonb** [comment]
I refreshed it again today as I am dogfooding with this while I develop my docs, and found something new that didn't work, (after the last change @dholbach mentioned which removed the previous thing that didn't work.)

I was able to fix it with a one-liner, and the organizational changes I made before were successful (eg. I didn't have any issue or get stuck compiling Hugo and golibsass all over again again just to update a simple thing in the `Dockerfile`.)

**Comment by dholbach** [comment]
I'm happy merging this - looks like it includes #274 as well now? @hiddeco do you want to take another look over the Makefile steps?

File: Makefile [hunk 2]
```diff
@@ -15,11 +15,13 @@ FONT_AWESOME_TARGET        := themes/$(DOCSY_COMMIT_FOLDER)/assets/vendor/$(FONT
...
 DEV_IMAGE_REGISTRY_NAME    := fluxcd
 HUGO_VERSION               ?= 0.78.2
-DEV_IMAGE_BASE_NAME        := website:hugo-$(HUGO_VERSION)-extended
-PREVIEW_IMAGE_BASE_NAME    := website:hugo-support
-DEV_IMAGE_NAME             := $(DEV_IMAGE_REGISTRY_NAME)/$(DEV_IMAGE_BASE_NAME)
-PREVIEW_IMAGE_NAME         := $(DEV_IMAGE_REGISTRY_NAME)/$(PREVIEW_IMAGE_BASE_NAME)
+HUGO_IMAGE_BASE_NAME       := website:hugo-$(HUGO_VERSION)-extended
+SUPPORT_IMAGE_BASE_NAME    := website:hugo-support
+HUGO_IMAGE_NAME            := $(DEV_IMAGE_REGISTRY_NAME)/$(HUGO_IMAGE_BASE_NAME)
+SUPPORT_IMAGE_NAME         := $(DEV_IMAGE_REGISTRY_NAME)/$(SUPPORT_IMAGE_BASE_NAME)
```
Comment (hiddeco) [inline Makefile:21]:
```suggestion
HUGO_IMAGE_NAME            ?= $(DEV_IMAGE_REGISTRY_NAME)/$(HUGO_IMAGE_BASE_NAME)
SUPPORT_IMAGE_NAME         ?= $(DEV_IMAGE_REGISTRY_NAME)/$(SUPPORT_IMAGE_BASE_NAME)
```

maybe, to allow for overwrites?

File: Makefile [hunk 3]
```diff
@@ -15,11 +15,13 @@ FONT_AWESOME_TARGET        := themes/$(DOCSY_COMMIT_FOLDER)/assets/vendor/$(FONT
 
 DEV_IMAGE_REGISTRY_NAME    := fluxcd
 HUGO_VERSION               ?= 0.78.2
```
Comment (dholbach) [inline Makefile:17]: This is at odds with https://github.com/fluxcd/website/blob/main/package.json#L4 - I think that's how netlify knows to use the extended version.

Comment (kingdonb) [reply Makefile:17]: I corrected the Makefile to use 0.83.1 matching package-lock.json

**Comment by kingdonb** [comment]
If there are concerns about merging #274 without much review, I can reorder the commits so this one gets merged by itself.

I am happy if we merge them both 👍

**Comment by kingdonb** [comment]
I also added an override for DEV_IMAGE_REGISTRY_NAME, which seems handy to me for pre-release testing

So you can do eg.

```
make docker-push docker-preview DEV_IMAGE_REGISTRY_NAME=kingdonb
```

for testing, which I just did (and it's LGTM)

**Comment by dholbach** [comment]
I don't really know anything about Jenkins, so have a bit of a hard time reviewing this...

**Comment by kingdonb** [comment]
That was the other notable omission from #274, the Jenkins guide... if you are not already a Jenkins user it might be important to show how to install Jenkins, but I left it out to avoid making this example too big and unfocused, not really trying to sell Jenkins to people on the street who haven't already decided to use it.

(snip: moved the Jenkins-specific comments over to #274)
Jenkins stuff is included in this PR but not the actual subject of the PR. Merging this will include both

This one simply fixes some stuff that broke in our `docker-support` image, and makes rebuilding it in the future easier with some refactors in the Makefile. I couldn't really effectively review and iterate on #274 without fixing the docker stuff, so that's how these PRs got mixed together in the first place.

**Comment by dholbach** [comment]
Can you rebase?

**Comment by dholbach** [comment]
@stefanprodan Can you run `make docker-push`?

**Comment by kingdonb** [comment]
@stefanprodan The version of `hugo` is changed, so you'll need to make sure `hugo/` directory is removed first before running again, if you have already cloned it at the previous version. I'm not sure how to solve this directly in the Makefile yet, (please be advised.)

**Pull request merged**